llm/falcon/README.md

```diff
@@ -0,0 +1,61 @@
+# Finetuning Falcon: The Top LLM Chatbot on OpenLLM Leaderboard
+
+This README contains instructions on how to use SkyPilot to finetune Falcon-7B and Falcon-40B, an open-source LLM that rivals many current closed-source models, including ChatGPT. Notably, it currently tops the charts of the [Open LLM Leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard).
+
+* [Blog post](https://huggingface.co/blog/falcon)
+* [Repo](https://huggingface.co/tiiuae/falcon-40b)
+* [Training code](https://gist.github.com/pacman100/1731b41f7a90a87b457e8c5415ff1c14)
+
+
+## Prerequisites
+Install the latest SkyPilot and check your setup of the cloud credentials:
+```bash
+pip install git+https://github.com/skypilot-org/skypilot.git
+sky check
+```
+See the Falcon SkyPilot YAML for [training](train.yaml). Serving is currently a work in progress and a YAML will be provided for that soon! We are also working on adding an evaluation step to evaluate the model you finetuned compared to the base model.
+
+## Finetuning Falcon with SkyPilot
+Finetuning Falcon-7B and Falcon-40B require GPUs with 80GB memory, but Falcon-7b-sharded requires only 40GB memory. In other words, if your GPU does not have over 40 GB memory, i.e. `accelerators: A100:1`, we suggest using `ybelkada/falcon-7b-sharded-bf16`. `tiiuae/falcon-7b` and `tiiuae/falcon-40b` will need 80 GB memory. In our examples, we use A100-80GB and A100. 
```

Comment:
I tried fine-tuning `ybelkada/falcon-7b-sharded-bf16` on A100-40GB. The training seemed to result in a very spiky loss. Is this expected:

<img width="431" alt="image" src="https://github.com/skypilot-org/skypilot/assets/4416605/958e78c5-b023-4835-90e7-fe1cab657a7c">

Reply:
I believe in the loss graph I sent you via Slack (below), I had different hyperparameters such as `max_seq_len` being a lot smaller. There are different sets of hyperparameters that work better for the different sized models, what should the default be in the YAML if that's the case?

<img width="546" alt="image" src="https://github.com/skypilot-org/skypilot/assets/32891260/7de9ea55-9844-4d1c-bc5a-e3613686593a">

Reply:
Hmm, we should try to include the hyperparameters if we know them. How about having something like this the `run` of the YAML:

if [ $MODEL_NAME == "tiiuae/falcon-7b" ]; then
    python train.py \
    --model_name $MODEL_NAME \
    --max_seq_len 4096 \
    --bf16 \
    --group_by_length \
    --bnb_4bit_compute_dtype bfloat16 \
    --max_steps 200 \
    --dataset_name timdettmers/openassistant-guanaco
  elif [ $MODEL_NAME == "tiiuae/falcon-40b" ]; then
    python train.py \
    --model_name $MODEL_NAME \
    --max_seq_len 2048 \
    --bf16 \
    --group_by_length \
    --bnb_4bit_compute_dtype bfloat16 \
    --max_steps 200 \
    --dataset_name timdettmers/openassistant-guanaco
  fi